azure/Guidelines.md

```diff
@@ -846,110 +847,259 @@ https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#perfo
+with an `operation-location` response header if the LRO Action operation was accepted for processing.
+
+<a href="#lro-put-action-returns-status-monitor" name="lro-put-action-returns-status-monitor">:white_check_mark:</a> **DO** return a
+status monitor in the response body that contains the operation status, request parameters, and when the operation completes either
+the operation result or error.
+
+Note: Since all request parameters must be present in the status monitor,
+the request and response body of the PUT can be defined with a single schema.
+
+<a href="#lro-put-action-status-monitor-url" name="lro-put-action-status-monitor-url">:ballot_box_with_check:</a> **YOU SHOULD**
+return the status monitor for an operation for a subsequent GET on the URL that initiates the LRO, and use this endpoint as
+the status monitor URL returned in the `operation-location` response header.
+
+#### The Status Monitor Resource
+
+All patterns that initiate a LRO either implicitly or explicitly create a [Status Monitor resource](https://datatracker.ietf.org/doc/html/rfc7231#section-6.3.3) in the service's `operations` collection.
+
+<a href="#lro-status-monitor-structure" name="lro-status-monitor-structure">:white_check_mark:</a> **DO** return a status monitor in the response body that conforms with the following structure:
 
 Property | Type        | Required | Description
 -------- | ----------- | :------: | -----------
 `id`     | string      | true     | The unique id of the operation
-`status` | string      | true     | enum that includes values "NotStarted", "Running", "Succeeded", "Failed", and "Canceled"
-`error`  | ErrorDetail |          | Error object that describes the error when status is "Failed"
-`result` | object      |          | Only for POST action-type LRO, the results of the operation when completed successfully
-additional<br/>properties | |     | Additional named or dynamic properties of the operation
+`kind`   | string enum | true(*)  | The kind of operation
+`status` | string enum | true     | The operation's current status: "NotStarted", "Running", "Succeeded", "Failed", and "Canceled"
+`error`  | ErrorDetail |          | If `status`=="Failed", contains reason for failure
+`result` | object      |          | If `status`=="Succeeded" && Action LRO (POST or PUT), contains success result if needed
+additional<br/>properties | | | Additional named or dynamic properties of the operation
 
-<a href="#lro-status-monitor-includes-all-fields" name="lro-status-monitor-includes-all-fields">:white_check_mark:</a> **DO** include the `id` of the operation and any other values needed for the client to form a GET request to the status monitor (e.g. a `location` path parameter).
+(*): When a status monitor endpoint supports multiple operations with different result structures or additional properties,
+the status monitor **must** be polymorphic -- it **must** contain a `kind` property that indicates the kind of long-running operation.
```

Comment:
Question: do we require the `kind` property, if the status monitor is only for a single type (resource or action)?

Reply:
No we don't. Is there a better wording here that would make this clearer?

Reply:
Thanks for the clarification.

Maybe just add a few words e.g. `(*) The kind property is required when a status monitor endpoint ...`